app/components/a_level_row_component.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class ALevelRowComponent < ViewComponent::Base
+  attr_reader :course, :errors
+
+  def initialize(course:, errors: nil)
+    super
+    @course = course
+    @errors = errors
+  end
+
+  def inset_text_css_classes
+    messages = errors&.values&.flatten
+
+    if messages&.include?('Enter degree requirements')
+      'app-inset-text--narrow-border app-inset-text--error'
+    else
+      'app-inset-text--narrow-border app-inset-text--important'
+    end
+  end
+
+  def has_errors?
+    inset_text_css_classes.include?('app-inset-text--error')
+  end
```

Comment:
```suggestion
    @errors = errors&.values&.flatten
    # or if we only care about the degree related error
    @errros = errors&.values.filter {  |err|  err == ''Enter degree requirements' }
  end

  def inset_text_css_classes
    if has_errors?
      'app-inset-text--narrow-border app-inset-text--error'
    else
      'app-inset-text--narrow-border app-inset-text--important'
    end
  end

  def has_errors?
     errors.any?
  end
```

It seems like we're doing a lot of gymnastics in this little class with with the errors.

Reply:
Yes, I agree. I copied and postponed what should I do with. At this moment A levels shouldn't have any this logic because the validation when publishing the course should be do in a separate card. (I won't create the card yet because I wanna talk with Kellie/Rose about it first). I simplified the class for now.